src/components/ToggleInput/ToggleInput.tsx

```diff
@@ -1,27 +1,42 @@
 import { InputTag } from "components/InputTag/InputTag";
 import { ToggleSwitch } from "components/ToggleSwitch/ToggleSwitch";
 import { useToggle } from "hooks/useToggle";
-import { DetailedHTMLProps, InputHTMLAttributes, useEffect, useState } from "react";
+import { DetailedHTMLProps, InputHTMLAttributes, useEffect } from "react";
 import style from "./ToggleInput.module.scss";
 
 type Props = {
-    label: string,
-    onToggle?: (state: boolean) => void,
-} & DetailedHTMLProps<InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>
+    id: string;
+    onChange: (state: number) => void;
+    onToggle: (state: boolean) => void;
+    disabled: boolean;
+} & Omit<
+    DetailedHTMLProps<InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>,
+    "onChange"
+>;
 
-export function ToggleInput({ onToggle, disabled, ...inputProps }: Props) {
-    const [isOn, setIsOn] = useState(false);
+export function ToggleInput({
+    onToggle,
+    onChange,
+    disabled,
+    ...inputProps
+}: Props) {
+    const [isOn, flip] = useToggle(!disabled);
 
-    const onSwitchToggle = (isSwitchOn: boolean) => {
-        setIsOn(isSwitchOn);
-        onToggle?.(isSwitchOn);
+    if (onToggle) {
```

Comment:
Hooks cant be conditional, they always have to execute. If you need the onToggle condition, put it inside the hook. I dont think you need the condition because onToggle its not optional.